### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by banigoden**
Create cicd for github Actions
Please check web_app-ci in git

### CODE REVIEW

File: .github/workflows/pre-commit.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      - '!master'
+      - '!main'
+      - '!refs/pull/*'
+
+jobs:
+  pre-commit:
+    runs-on:
+      - self-hosted
+    container:
+      image: ghcr.io/lek-x/ubuntu_lint:v2
```
Comment (lek-x) [inline .github/workflows/pre-commit.yaml]: don't remember that i expoused my image to public

File: .github/workflows/pre-commit.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+          pip install pre-commit
+          pre-commit install
+
+      - name: Run linters
+        run: pre-commit run --all-files
+
+      - name: Set up Decker Buildx
+        uses: docker/login-action@v2
+
+      - name: Login to DockerHub
```
Comment (lek-x) [inline .github/workflows/pre-commit.yaml]: not here, this action only for lint

File: .github/workflows/pre-commit.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Build and push Docker image
```
Comment (lek-x) [inline .github/workflows/pre-commit.yaml]: same as previous

File: .github/workflows/web_app-ci.yaml [hunk 4]
```diff
@@ -0,0 +1,63 @@
+---
```
Comment (lek-x) [inline .github/workflows/web_app-ci.yaml]: name this workflow  e.g "Deploy"

File: .github/workflows/web_app-ci.yaml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+env:
+  AWS_REGION: 'us-west-2'
+  IMAGE_NAME: 'banigoden/web_app'
+  IMAGE_VERSION: ${{ github.run_number }}
+  BRANCH_NAME: ${{ github.ref_name }}
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
```
Comment (lek-x) [inline .github/workflows/web_app-ci.yaml]: let's make first steps for buildiong and publishing docker, after terraform and anible etc.

**Review by lek-x — CHANGES_REQUESTED** [review]

File: .github/workflows/web_app-ci.yaml (hunk 5 above)
Comment (banigoden) [reply .github/workflows/web_app-ci.yaml]: moved from pre-commit to deploy

**Pull request merged**